File: articles/the-shadow-and-the-psychology-of-disowned-selfhood/cpp/shadow_projection_scores.cpp
/*
The Shadow and the Psychology of Disowned Selfhood
C++ example: simple synthetic shadow-projection score
*/

#include <cmath>
#include <iomanip>
#include <iostream>

double shadow_activation(
    double latent_disowned,
    double cue_intensity,
    double shadow_discrepancy,
    double affective_charge,
    double reflective_capacity
) {
    return 0.72 * latent_disowned * cue_intensity
         + 0.56 * shadow_discrepancy
         + 0.42 * affective_charge
         - 0.48 * reflective_capacity;
}

double projection_intensity(
    double shadow_activation_value,
    double affective_charge,
    double persona_rigidity,
    double reflective_capacity
) {
    return 0.76 * shadow_activation_value
         + 0.44 * affective_charge
         + 0.32 * persona_rigidity
         - 0.56 * reflective_capacity;
}

double integration_capacity(
    double reflective_capacity,
    double projection_intensity_value,
    double shame_response,
    double developmental_time
) {
    return 0.50 * reflective_capacity
         - 0.34 * std::fabs(projection_intensity_value)
         - 0.26 * shame_response
         + 0.28 * developmental_time;
}

int main() {
    double shadow = shadow_activation(0.58, 0.46, 0.02, 0.48, 0.74);
    double projection = projection_intensity(shadow, 0.48, 0.52, 0.74);
    double integration = integration_capacity(0.74, projection, 0.24, 1.0);

    std::cout << std::fixed << std::setprecision(3);
    std::cout << "Synthetic shadow activation: " << shadow << std::endl;
    std::cout << "Synthetic projection intensity: " << projection << std::endl;
    std::cout << "Synthetic integration capacity: " << integration << std::endl;
    return 0;
}

File: articles/the-shadow-and-the-psychology-of-disowned-selfhood/fortran/shadow_projection_scores.f90
! The Shadow and the Psychology of Disowned Selfhood
! Fortran example: simple synthetic shadow-projection score

program shadow_projection_scores_demo
  implicit none

  real :: shadow
  real :: projection
  real :: integration

  shadow = 0.72 * 0.58 * 0.46 + 0.56 * 0.02 + 0.42 * 0.48 - 0.48 * 0.74
  projection = 0.76 * shadow + 0.44 * 0.48 + 0.32 * 0.52 - 0.56 * 0.74
  integration = 0.50 * 0.74 - 0.34 * abs(projection) - 0.26 * 0.24 + 0.28 * 1.0

  print *, "Synthetic shadow activation:", shadow
  print *, "Synthetic projection intensity:", projection
  print *, "Synthetic integration capacity:", integration
end program shadow_projection_scores_demo
